File: Hearthstone-Script/src/main/java/club/xiaojiawei/utils/RandomUtil.java
package club.xiaojiawei.utils;

import lombok.Getter;

import java.util.Random;

/**
 * 随机数生成工具
 * @author 肖嘉威
 * @date 2022/11/24 19:41
 */
public class RandomUtil {

    @Getter
    private static final Random RANDOM = new Random(System.currentTimeMillis());

    public static int getRandom(int min, int max){
        if (min > max){
            return getRandom(max, min);
        }
        return (int) (RANDOM.nextDouble() * (max - min + 1) + min);
    }

    public static double getRandom(double min, double max){
        if (min > max){
            return getRandom(max, min);
        }
        return RANDOM.nextDouble(min, max);
    }

    public static int getHugeRandom(){
        return getRandom(3000, 5000);
    }
    public static int getLongRandom(){
        return getRandom(2000, 3000);
    }
    public static int getMediumRandom(){
        return getRandom(1000, 1600);
    }
    public static int getShortMediumRandom(){
        return getRandom(500, 800);
    }
    public static int getShortRandom(){
        return getRandom(250, 450);
    }
    public static int getTinyRandom(){
        return getRandom(100, 200);
    }
    public static int getHumanRandom(){
        return getRandom(200, 2000);
    }
}

File: Hearthstone-Script-Base/src/main/kotlin/club/xiaojiawei/util/RandomUtil.kt
package club.xiaojiawei.util

import java.util.*

/**
 * 随机数生成工具
 * @author 肖嘉威
 * @date 2022/11/24 19:41
 */
object RandomUtil {

    private val RANDOM = Random(System.currentTimeMillis())

    fun getRandom(min: Int, max: Int): Int {
        if (min == max) return min
        return if (min > max) {
            getRandom(max, min)
        } else (RANDOM.nextDouble() * (max - min + 1) + min).toInt()
    }

    fun getRandom(min: Double, max: Double): Double {
        if (min == max) return min
        return if (min > max) {
            getRandom(max, min)
        } else RANDOM.nextDouble(min, max)
    }

    fun getHugeRandom(): Int {
        return getRandom(3000, 5000)
    }

    fun getLongRandom(): Int {
        return getRandom(2000, 3000)
    }

    fun getMediumRandom(): Int {
        return getRandom(1000, 1600)
    }

    fun getShortMediumRandom(): Int {
        return getRandom(500, 800)
    }

    fun getShortRandom(): Int {
        return getRandom(250, 450)
    }

    fun getTinyRandom(): Int {
        return getRandom(100, 200)
    }

    fun getHumanRandom(): Int {
        return getRandom(200, 2000)
    }

}
